tests/Feature/CreatePaymentTest.php

```diff
@@ -209,3 +213,129 @@
+    $mollieMockPayment->id = uniqid('tr_');
+    $mollieMockPayment->status = PaymentStatus::STATUS_PAID;
+    $mollieMockPayment->method = PaymentMethod::IDEAL;
+    $mollieMockPayment->amount = [
+        'value' => '20.00',
+        'currency' => 'USD',
+    ];
+
+    $mollieMockPayment->_links = [
+        'checkout' => [
+            'href' => 'https://www.mollie.com/checkout/test-mode?method=ideal&token=6.5gwscs',
+        ],
+    ];
+
+    Http::fake([
+        'https://api.mollie.com/*' => Http::response(json_encode($mollieMockPayment)),
+    ]);
+
+    Currency::factory()->create([
+        'code' => 'USD',
+        'decimal_places' => 4,
+    ]);
+
+    $amount = MollieManager::normalizeAmountToInteger('20.00', 'USD');
+
+    $response = $this
+        ->jsonApi()
+        ->expects('orders')
+        ->withData([
+            'type' => 'orders',
+            'id' => (string) $this->order->getRouteKey(),
+            'attributes' => [
+                'payment_method' => 'mollie',
+                'amount' => $amount,
+                'meta' => [
+                    'payment_method_type' => PaymentMethod::IDEAL,
+                    'payment_method_issuer' => 'ideal_ABNANL2A',
+                ],
+            ],
+        ])
+        ->post($url);
+
+    $response->assertSuccessful();
+
+    // Expect a checkout url to be returned
+    expect($response->json('meta.payment_intent.meta.mollie_checkout_url'))->toBeString();
+
+    // Expect a transaction to be created
+    $this->assertDatabaseHas('lunar_transactions', [
+        'order_id' => $this->order->id,
+        'card_type' => PaymentMethod::IDEAL,
+        'amount' => $amount,
+    ]);
+});
+
+test('can normalize amount to integer', function () {
+    $currency = Currency::getDefault();
+
+    $amount = MollieManager::normalizeAmountToInteger('20.00', $currency->code);
+    expect($amount)->toBeInt();
+    expect($amount)->toBe(2000);
+
+    $amount = MollieManager::normalizeAmountToInteger('20.96', $currency->code);
+    expect($amount)->toBeInt();
+    expect($amount)->toBe(2096);
+
+    $amount = MollieManager::normalizeAmountToInteger('20.60', $currency->code);
+    expect($amount)->toBeInt();
+    expect($amount)->toBe(2060);
+
+    $amount = MollieManager::normalizeAmountToInteger('20.6', $currency->code);
+    expect($amount)->toBeInt();
+    expect($amount)->toBe(2060);
+
+    $amount = MollieManager::normalizeAmountToInteger('20', $currency->code);
+    expect($amount)->toBeInt();
+    expect($amount)->toBe(2000);
+
+    $amount = MollieManager::normalizeAmountToInteger('20.00', $currency->code);
+    expect($amount)->toBeInt();
+    expect($amount)->toBe(2000);
+});
+
+test('can normalize amount to integer for a currency with 4 decimals', function () {
+    $currency = Currency::factory()->create([
+        'code' => 'USD',
+        'decimal_places' => 4,
+    ]);
+
+    $amount = MollieManager::normalizeAmountToInteger('20.00', $currency->code);
```

Comment:
Is er geen test nodig met een float/decimal?

Reply:
Nee, ik sta alleen strings toe en ben nog geen situatie tegengekomen waar het geen string was.